index.bs

```diff
@@ -16,13 +16,443 @@ WPT Display: closed
 
-For now, see the [explainer]([REPOSITORYURL]).
+The [Storage Access API](https://github.com/privacycg/storage-access) supports "authenticated embeds" by providing a way to request access to unpartitioned cookies in an embedded context. This currently requires an explicit call to a ECMAScript API (namely {{Document/requestStorageAccess()}}) to:
+1. Potentially prompt the user for permission; and
+1. Explicitly indicate the embedded resource's interest in using unpartitioned cookies (as a protection against CSRF attacks by an embedder).
+
+As the above list suggests, this single API invocation is serving two orthogonal purposes:
+1. It enforces a privacy boundary between the top-level site and the embedded site, and gives the user (and/or user agent) an opportunity to relax or maintain that privacy boundary.
+1. It enforces a security boundary between the top-level site and the embedded site (namely, the aforementioned CSRF protection), and serves as the embedded site's explicit signal to relax that security boundary (by allowing credentialed requests to be sent to the embedded site, in the given context).
+
+The requirement to invoke {{Document/requestStorageAccess}} is therefore useful, but it imposes some challenges:
+* Use of the Storage Access API may currently require multiple network round trips and multiple resource reloads before an <{iframe}> can work as expected, since the <{iframe}> must execute {{Document/requestStorageAccess()}} before fetching all of its embedded resources (presuming that they require access to cross-site cookies). In practice this means that the <{iframe}> loads, executes {{Document/requestStorageAccess()}}, then refreshes itself in order to re-do all of the embedded fetches (including cross-site cookies this time).
+* Embedded resources currently must execute ECMAScript in order to benefit from this API. This effectively means that the embedded resource must be an <{iframe}> that has the ability to run ECMAScript, or must be a subresource fetched by such an <{iframe}> (see the first bullet). This imposes an unnecessary burden on sites that serve access-controlled resources (i.e. resources that require authentication cookies) which are embedded in cross-site pages.
+
+These challenges can be mitigated by supporting a new pair of headers. In particular, this document introduces:
+
+* [:Sec-Fetch-Storage-Access:], a request header to convey information about whether cross-site cookies were included in the request, and possibly whether the <a permission><code>storage-access</code></a> permission has been granted.
+* [:Activate-Storage-Access:], a response header that can be used to activate an existing <a permission><code>storage-access</code></a> permission grant and "retry" the request, or to activate an existing <a permission><code>storage-access</code></a> prior to loading a [=Document=] (typically, an <{iframe}>).
+
+Infra {#infra}
+==============
+
+This specification depends on the Infra standard. [[!INFRA]]
```

Comment:
Why is this section needed? I don't see it in other specs.

Reply:
I took this section from https://privacycg.github.io/storage-access/#infra, since the Infra Standard declares that specs must explicitly declare their dependency on it (https://infra.spec.whatwg.org/#usage).

Reply:
OK I suppose that makes sense. Didn't realize that requirement existed (it seems loosely enforced). Resolving this sounds good.